webpush/encrypt.go

```diff
@@ -55,13 +55,43 @@ import (
 	"strings"
 )
 
+// ContentEncoding indicates the version of encoding.
+type ContentEncoding int
+
 const (
-	maxPayloadLength = 4078
+	// The encoding that is widely deployed with WebPush (as of 2016-11).
+	AESGCM ContentEncoding = iota
+	// The most recent encoding, the salt, record size and key identifier
+	// are included in a header that is part of the encrypted content coding.
+	AES128GCM
+)
+
+func (v ContentEncoding) String() string {
+	switch v {
+	case AESGCM:
+		return "aesgcm"
+	case AES128GCM:
+		return "aes128gcm"
+	}
+	panic("A valid content encoding must be used")
```

Comment:
Well, you can always pass something else if the base type is exported or built-in, like in this case.
Here's an example: https://play.golang.org/p/m-ffT2LvBH

You'd have to make ContentEncoding from some unexported type, but I don't think it's really worth it.